Simplify and make js manual doc more correct (#1435)
open-telemetry/opentelemetry.io: content/en/docs/instrumentation/js/instrumentation.md

## Initializing a Tracer

> As you have learned in the previous [Getting Started](../getting-started) guide

Reviewer: User may not have read the previous guide.

you need a TracerProvider and an Exporter. Install the dependencies and add them
to the head of your application code to get started:

```shell
npm install @opentelemetry/api
npm install @opentelemetry/sdk-trace-base
```

Next, initialize a tracer, preferably in a separate file (e.g.,
`instrumentation-setup.js`):

```javascript
const {
  BasicTracerProvider,
  ConsoleSpanExporter,

>   BatchSpanProcessor,

Reviewer: Batch span processor makes sense in some cases (typical server app) but not in others (serverless). It might make sense to add a comment about this because a serverless container may be killed before a batch is exported.

);
```

> This registers a tracer provider with the OpenTelemetry API as the global tracer
> provider, and exports a tracer instance that you can use to create spans.
> 
> If you do not register a global tracer provider, any instrumentation calls will
> be a no-op, so this is important to do!

Reviewer: We want to encourage using the API and not using SDK objects directly.

```suggestion
This registers a global `TracerProvider` with the OpenTelemetry API, which may be used to create spans.
If no global `TracerProvider` is registered, instrumentations which use the API will use a no-op tracer and no spans will be created or exported.
```

## Create nested spans

Nested spans let you track work that's nested in nature. For example, the
`doWork` function below represents a nested operation. The following sample
creates a nested span that tracks the `doWork` function:

```javascript
const mainWork = () => {
  const parentSpan = tracer.startSpan('main');

  for (let i = 0; i < 3; i += 1) {
    doWork(parentSpan, i);
  }

  // Be sure to end the parent span!
  parentSpan.end();
}

function doWork(parent, i) {

>   // To set a child span, we need to mark the parent span as the active span

Reviewer: Not obvious what "set a child span" means for those not already very familiar with tracing.

Finally, you can update your file to include semantic attributes:

```javascript
const doWork = () => {
  const span = tracer.startSpan(
    'app.new-span',
    { attributes: { [SemanticAttributes.CODE_FUNCTION]: 'doWork' } },
    ctx
  );

  // Add an attribute to the same span later on
  span.setAttribute(SemanticAttributes.CODE_FILEPATH, __filename);

  // Make sure to end the span if you're done with it!
  span.end();
}
```

## Span events

An event is a human-readable message attached to a span that represents

> "something happening" during its lifetime. You can think of it like a primitive

Reviewer: IMO "something happening" is too general. I prefer to think of events as an activity that is tracked as a single point in time rather than a start time and and end time. For example, a page load may take time and be represented by a span, but the moment a page becomes interactive is a singular point in time and may be represented as a span event.

## Span Status

A status can be set on a span, typically used to specify that there was an error
in the operation a span is tracking - .`Error`.

```javascript
const span = tracer.startSpan('doWork', undefined, ctx);

for (let i = 0; i <= Math.floor(Math.random() * 40000000); i += 1) {
  if (i > 10000) {
    span.setStatus({
      code: opentelemetry.SpanStatusCode.ERROR,
      message: 'Error'
    });
  }
}

span.end();
```

By default, the status for all spans is `Unset`. In rare cases, you may also

> wish to set the status to `Ok`. This should generally not be necessary, though.

Reviewer: Might be a good idea to explain why this is not necessary?
